### PR TITLE
NAS-110803 / 21.06 / Do not treat mountpoint specially

### PR DESCRIPTION
**Pull request by sonicaj**
This PR introduces changes to make sure we are consistent in our usage wrt consuming `mountpoint` property.
Before py-libzfs used to return `mountpoint` key/value always even if it did or did not return the actual value of the property. This can be confusing and lead to errors/mistakes - the changes introduced here expect the key to not be present anymore and instead rely on the property to be retrieved explicitly and not make an implicit expectation that it's going to be there.

### CODE REVIEW

**Review by themylogin — APPROVED** [review]

File: src/middlewared/middlewared/plugins/pool.py [hunk 1]
```diff
@@ -2863,6 +2864,12 @@ def transform(dataset):
                 if method:
                     dataset[i]['value'] = method(dataset[i]['value'])
 
+            if 'mountpoint' in dataset:
+                # This is treated specially to keep backwards compatibility with API
+                dataset['mountpoint'] = dataset['mountpoint']['value']
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/pool.py:2869]: I think mountpoint should always be there, we cannot have top level properties that may or may not be there, its very misleading and backward incompatible.

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/pool.py:2869]: So if mountpoint property was not retrieved explicitly, what should we say for `mountpoint` property ?

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/pool.py:2869]: Ability to specify custom properties was introduced recently - which means that for old users/releases, `mountpoint` property would continue to exist always

Comment (william-gr) [reply src/middlewared/middlewared/plugins/pool.py:2869]: Recently as in SCALE? not 12?

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/pool.py:2869]: @william-gr, yes that is accurate - for `pool.dataset.query`, retrieving specific properties was introduced in SCALE

Comment (william-gr) [reply src/middlewared/middlewared/plugins/pool.py:2869]: Ok, then sounds reasonable, thanks for clarifying!

**Review by william-gr — APPROVED** [review]

**Pull request merged**